### ISSUE TITLE
Cannot schema evolve a union of a logical type

### ISSUE THREAD

**Issue by tboothman** [mtth/avsc#151]
You can schema evolve a logical type to the same logical type without a problem, but if you try a logical type in a union it fails.

The problem seems to be where the _resolve method is called on the logical type. In a union the type parameter is an instance of the underlying type rather than the logical type ... but outside of a union it's the logical type.

```javascript
const Avsc = require("avsc");

class DateTimeType extends Avsc.types.LogicalType {
    _fromValue(val) {
        return new Date(val);
    }
    _toValue(date) {
        return date instanceof Date ? +date : undefined;
    }
    _resolve(type) {
        console.log(type);
        if (Avsc.Type.isType(type, 'logical:timestamp-millis')) {
            return this._fromValue;
        }
    }
}
const schemaData = {
    "type": "record",
    "fields": [
        {
            "name": "n",
            "type": [
                "null",
                {
                    "type": "long",
                    "logicalType": "timestamp-millis"
                }
            ]
        }
    ]
};
let schema = Avsc.Type.forSchema(schemaData, {
    logicalTypes: {
        "timestamp-millis": DateTimeType
    }
});
schema.createResolver(schema);
```

```
$ node dist/avro.js
<LongType>
/code/node_modules/avsc/lib/types.js:527
throw new Error(f('cannot read %s as %s', type, this));
^

Error: cannot read "long" as ["null","long"]
at UnwrappedUnionType.Type.createResolver (/code/node_modules/avsc/lib/types.js:527:11)
at UnwrappedUnionType.Type.createResolver (/code/node_modules/avsc/lib/types.js:486:17)
at UnwrappedUnionType.<anonymous> (/code/node_modules/avsc/lib/types.js:512:19)
at Array.map (<anonymous>)
at UnwrappedUnionType.Type.createResolver (/code/node_modules/avsc/lib/types.js:511:32)
at RecordType._update (/code/node_modules/avsc/lib/types.js:2334:30)
at RecordType.Type.createResolver (/code/node_modules/avsc/lib/types.js:523:10)
at Object.<anonymous> (/code/dist/avro.js:39:8)
at Module._compile (module.js:624:30)
at Object.Module._extensions..js (module.js:635:10)
```
Note the `<LongType>` in the console.log from _resolve.

This isn't the actual code I'm using, I trimmed it down to make it simpler - I know your example datetime-millis allows the type to be a long or string which makes the evolution work. I'm using a decimal type which needs to see and compare the scale and precision to allow the schema resolution to work.

**Comment by mtth** [comment]
Thanks for the trimmed down example; there was a indeed resolution bug when a logical type was read as logical type nested inside a union. It should be fixed in https://github.com/mtth/avsc/commit/e0ed272c3eb51a358461296af7c2fa9eed6b6588, released in `5.2.1` alongside https://github.com/mtth/avsc/pull/150. (In this release, your example correctly outputs `<LogicalType {"type":"long","logicalType":"timestamp-millis"}> ` and terminates successfully.)

**Comment by tboothman** [comment]
Acknowledged, fixed and released in less than a day .. that's pretty amazing.
Thanks a lot @mmth - I may have run into two bugs trying to resolve one schema but you made that a pretty painless experience.
